packages/rollup-core/src/app/data/producers/l2-chain-data-persister.ts

```diff
@@ -135,6 +139,7 @@ export class L2ChainDataPersister extends ChainDataProcessor {
       stateRoot: block['stateRoot'], // should be added by rollup-core/app/utils.ts: monkeyPatchL2Provider
       gasLimit: L2ChainDataPersister.parseBigNumber(response.gasLimit),
       gasPrice: L2ChainDataPersister.parseBigNumber(response.gasPrice),
+      type: (response as OptimismTransactionResponse).type,
     }
```

Comment:
Think it would be cleaner to replace `TransactionOutput` on line 130 with `OptimismTransactionResponse`? This also depends on the type field being returned from geth2

Reply:
I made `OptimismTransactionResponse` extend `TransactionOutput` here: https://github.com/ethereum-optimism/optimism-monorepo/pull/261/files#diff-720226648298c371f0a15b5f1f9d371cR150

My intention was to make `OptimismTransactionResponse` a superset of `TransactionOutput` and add a single new field `type`. Updates to the geth2 RPC need to happen and also there could be more fields on the `OptimismTransactionResponse`, basically the things in the `txmeta`.

Reply:
That extends ethers' `TransactionResponse` not `TransactionOutput` right? If switched to `TransactionOutput`, I think we're on the same page 👍

Reply:
Yes you are right @willmeister, my bad on the typo.

```js
export interface OptimismTransactionResponse extends TransactionResponse {
  type: string
} 
```